polyfactory/factories/base.py

```diff
@@ -805,7 +805,18 @@ def get_field_value_coverage(  # noqa: C901
                 )
 
             elif (origin := get_type_origin(unwrapped_annotation)) and issubclass(origin, Collection):
-                yield handle_collection_type_coverage(field_meta, origin, cls)
+                if not field_meta.children:
+                    msg = "A subclass of Collection should always have children in it's field_meta"
+                    raise ParameterException(msg)
+
+                # We actually want to use the parent in cases where the collection is the parent
+                # such as tuple so default to the parent if this annotation is not present in the children
```

Comment:
Sorry, but I didn't understand this. Could you explain this with an example? This is just me not understanding :D

Reply:
It took me a while to figure this out myself. I would really appreciate it if you have a better way to word that comment from your second hand perspective.

There are 2 kinds of collection for the purposes of coverage,  a collection because the type to be covered is a collection, and a collection because there are multiple things that can fill a type. If you don't disambiguate them, you end up creating an unintended collection nesting because there was a collection of things the collection could have been. 

When processing the `CoverageContainer` for `Foo`
```python
class Foo():
   bar: tuple[int, int] | str
   baz: tuple[int, int]
```
The structure of the options for filling Foo will look something like this
```
bar = [                         # Container for the ways bar can be filled
   tuple[int, int]          # The first way to fill bar is with a tuple containing...
   str,                            # The other way to fill bar
]
baz = tuple[int, int]  # The only way to fill baz, which happens to be a container
```
`bar` has `tuple[int, int]` and `str` as children in it's `field_meta`. 
`baz` has `int` and `int` as children in it's `field_meta`. 

When we are figuring out what values to put into bar we; 
1. loop through the unwrapped annotation of bar
2. get to filling the collection annotation `tuple[int, int]`
3. **find the `FieldMeta` associated with that collection in the children of bar**
4. create coverage for that `FieldMeta` with the type `origin` of `tuple`

When we are figuring out what values to put into baz we;
1. loop through the unwrapped annotation of baz
2. get to filling the collection annotation `tuple[int, int]`
3. **Don't find the `FieldMeta` associated with that collection in the children of baz because its' children are `int` and `int`**
4. **Default to the `FieldMeta` of baz which actually represents the collection**
5. create coverage for that `FieldMeta` with the type `origin` of `tuple`

Reply:
Aah gotcha, that makes sense :) I think it's probably enough to add a reference to this PR for more details as part of the comment. With this explanation, the comment makes sense to me.